ietfYangDraftPull/draftPullLocal.py

```diff
@@ -72,96 +75,28 @@ def get_help(self):
         return ret
 
 
-def get_latest_revision(f, LOGGER):
-    """
-    Search for revision in yang file
-    :param f: yang file
-    :return: revision of the file "f"
-    """
-    stmt = yangParser.parse(f)
-    if stmt is None: # In case of invalid YANG syntax, None is returned
-        LOGGER.info('Cannot yangParser.parse ' + f)
-        return None
-    rev = stmt.search_one('revision')
-    if rev is None:
-        return None
-
-    return rev.arg
+def run_populate_script(directory: str, notify: bool, LOGGER):
+    """ Run populate.py script and return whether execution was successful or not.
 
-
-def check_name_no_revision_exist(directory, LOGGER_temp):
-    """
-    This function checks the format of all the modules filename.
-    If it contains module with a filename without revision,
-    we check if there is a module that has revision in
-    its filename. If such module exists, then module with no revision
-    in filename will be removed.
-    :param directory: (str) path to directory with yang modules
+    Argumets:
+        :param directory    (str) full path to directory with yang modules
+        :param notify       (str) whether to send files for'indexing
+        :param LOGGER       (obj) formated logger with the specified name
     """
-    LOGGER = LOGGER_temp
-    LOGGER.debug('Checking revision for directory: {}'.format(directory))
-    for root, dirs, files in os.walk(directory):
-        for basename in files:
-            if '@' in basename:
-                yang_file_name = basename.split('@')[0] + '.yang'
-                revision = basename.split('@')[1].split('.')[0]
-                exists = os.path.exists(directory + yang_file_name)
-                if exists:
-                    compared_revision = get_latest_revision(os.path.abspath(directory + yang_file_name), LOGGER)
-                    if compared_revision is None:
-                        continue
-                    if revision == compared_revision:
-                        os.remove(directory + yang_file_name)
+    successful = True
+    try:
+        module = __import__('parseAndPopulate', fromlist=['populate'])
+        submodule = getattr(module, 'populate')
+        script_conf = submodule.ScriptConfig()
+        script_conf.args.__setattr__('sdo', True)
+        script_conf.args.__setattr__('dir', directory)
+        script_conf.args.__setattr__('notify_indexing', notify)
+        submodule.main(scriptConf=script_conf)
+    except Exception as e:
+        LOGGER.error('populate script error:\n{}'.format(e))
```

Comment:
LOGGER.exception instead of error. This will give us more information and traceback. https://docs.python.org/3/library/logging.html

you should be using that in any
 ```
except  Exception as e:
    ...
```
it should not be used outside of except clause

Reply:
Done